### PR TITLE
Fix strategy to remain the same in all tasks if chosen in a task

### PR DESCRIPTION
**Pull request by msmner**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/476

### CODE REVIEW

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx [hunk 1]
```diff
@@ -106,17 +106,25 @@ const SubmissionsProvider = ({ children }: ISubmissionsProviderProps) => {
...
 
             if (submissionType) {
                 const { id } = submissionType;
 
                 selectSubmissionTypeById(id);
             } else {
-                selectSubmissionTypeById(null);
+                const defaultSubmissionType = first(allowedSubmissionTypes);
+
+                if (defaultSubmissionType) {
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx:118]: No nested `if`s

**Review by Minkov — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx (hunk 1 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx]: Fixed.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx [hunk 2]
```diff
@@ -106,17 +106,21 @@ const SubmissionsProvider = ({ children }: ISubmissionsProviderProps) => {
...
-            if (submissionType) {
-                const { id } = submissionType;
+            if (!submissionType) {
+                const defaultSubmissionType = first(allowedSubmissionTypes);
 
-                selectSubmissionTypeById(id);
-            } else {
-                selectSubmissionTypeById(null);
+                return isNil(defaultSubmissionType)
+                    ? selectSubmissionTypeById(null)
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx:115]: should spacings appear like this?

**Review by shsimeonova — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx [hunk 3]
```diff
@@ -106,17 +106,21 @@ const SubmissionsProvider = ({ children }: ISubmissionsProviderProps) => {
     useEffect(
         () => {
             const { allowedSubmissionTypes } = currentProblem || {};
-            const submissionType = first(allowedSubmissionTypes);
+            const submissionType = allowedSubmissionTypes?.find((st) => st.id === selectedSubmissionType?.id);
 
-            if (submissionType) {
-                const { id } = submissionType;
+            if (!submissionType) {
```
Comment (mishogv) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx:111]: use `!IsNil(submissionType)`

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx [hunk 4]
```diff
@@ -106,17 +106,21 @@ const SubmissionsProvider = ({ children }: ISubmissionsProviderProps) => {
...
+                const defaultSubmissionType = first(allowedSubmissionTypes);
 
-                selectSubmissionTypeById(id);
-            } else {
-                selectSubmissionTypeById(null);
+                return isNil(defaultSubmissionType)
+                    ? selectSubmissionTypeById(null)
+                    : selectSubmissionTypeById(defaultSubmissionType.id);
```
Comment (mishogv) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx:116]: destruct defaultSubmissionType.id

**Review by mishogv — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx (hunk 3 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx:111]: Done.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx (hunk 4 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx:116]: When I try to destructure it before the check if defaultSubmissionType is nil, it says the id does not exist on type ISubmissionTypeType | undefined. So I need to check first if the default submission type is undefined to destructure it afterwards, but I don't see a way to do it without involving a nested if.

**Review by mishogv — APPROVED** [review]

**Review by shsimeonova — APPROVED** [review]

**Review by mishogv — APPROVED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx [hunk 5]
```diff
@@ -146,17 +146,21 @@ const SubmissionsProvider = ({ children }: ISubmissionsProviderProps) => {
...
             }
+
+            const { id } = submissionType;
+
+            return selectSubmissionTypeById(id);
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx:161]: The whole `useEffect` can be done as follows:

```
const { allowedSubmissionTypes } = currentProblem || {};
const submissionType = allowedSubmissionTypes?.find((st) => st.id === selectedSubmissionType?.id);

const { id } = submissionType || first(allowedSubmissionTypes) || { id: null };
selectSubmissionTypeById(id);
```

**Review by Minkov — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx (hunk 5 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx:161]: Great, refactored!

**Review by mishogv — APPROVED** [review]

**Review by Minkov — APPROVED** [review]

**Pull request merged**